src/schema/v2/searchCriteriaLabel.ts

```diff
@@ -0,0 +1,346 @@
+    },
+    value: {
+      type: GraphQLString,
+      description: "The human-friendly value of the filter facet",
+    },
+  },
+})
+
+/**
+ * A resolver that takes the current SearchCriteria and returns a list of
+ * SearchCriteriaLabels representing that SearchCriteria in human-friendly form
+ */
+export const resolveSearchCriteriaLabels = async (
+  parent,
+  _args,
+  context,
+  _info
+) => {
+  const {
+    artistIDs,
+    attributionClass,
+    additionalGeneIDs,
+    priceRange,
+    sizes,
+    width,
+    height,
+    acquireable,
+    atAuction,
+    inquireableOnly,
+    offerable,
+    materialsTerms,
+    locationCities,
+    majorPeriods,
+    colors,
+    partnerIDs,
+  } = parent
+
+  const { artistLoader, partnerLoader } = context
+
+  const labels: any[] = []
+
+  labels.push(await getArtistLabels(artistIDs, artistLoader))
+  labels.push(getRarityLabels(attributionClass))
+  labels.push(getMediumLabels(additionalGeneIDs))
+  labels.push(getPriceLabel(priceRange))
+  labels.push(getSizeLabels(sizes))
+  labels.push(getCustomSizeLabels({ width, height }))
+  labels.push(
+    getWaysToBuyLabels({
+      acquireable,
+      atAuction,
+      inquireableOnly,
+      offerable,
+    })
+  )
+  labels.push(getMaterialLabels(materialsTerms))
+  labels.push(getLocationLabels(locationCities))
+  labels.push(getPeriodLabels(majorPeriods))
+  labels.push(getColorLabels(colors))
+  labels.push(await getPartnerLabels(partnerIDs, partnerLoader))
+
+  return labels.flat().filter((x) => x !== undefined) as SearchCriteriaLabel[]
+}
+
+async function getArtistLabels(artistIDs: string[], artistLoader) {
+  if (!artistIDs?.length) return []
+
+  return Promise.all(
+    artistIDs.map(async (id) => {
+      const artist = await artistLoader(id)
+      return {
+        name: "Artist",
+        value: artist.name,
+        field: "artistIDs",
+      }
+    })
+  )
+}
+
+function getRarityLabels(attributionClasses: string[]) {
+  if (!attributionClasses?.length) return []
+
+  return attributionClasses.map((attributionClass) => ({
+    name: "Rarity",
+    value: allAttributionClasses[attributionClass].name,
+    field: "attributionClass",
+  }))
+}
+
+function getMediumLabels(additionalGeneIDs: string[]) {
+  if (!additionalGeneIDs?.length) return []
+
+  // Corresponds to the list of options under
+  // the Medium facet on an artwork grid.
+  //
+  // Being a list of genes, this is related to,
+  // but not the same as, the list of artwork medium types.
+
+  const MEDIUM_GENES = {
+    painting: "Painting",
+    photography: "Photography",
+    sculpture: "Sculpture",
+    prints: "Prints",
+    "work-on-paper": "Work on Paper",
+    nft: "NFT",
+    design: "Design",
+    drawing: "Drawing",
+    installation: "Installation",
+    "film-slash-video": "Film/Video",
+    jewelry: "Jewelry",
+    "performance-art": "Performance Art",
+    reproduction: "Reproduction",
+    "ephemera-or-merchandise": "Ephemera or Merchandise",
+  }
```

Comment:
During an earlier knowledge share, I showed a version where gene names (corresponding to the Medium filter) were _also_ being fetched from Gravity.

But it makes sense to do those with a static lookup here, since there are only a dozen or so possibilities (unlike artists and partners which are unbounded collections).

Reply:
Yeah, that sounds reasonable to me. This did make me think about whether we could further consolidate our filter logic in MP in the future - basically doing what's in this PR for our filter menus to avoid clients like [Force doing a similar conversion](https://github.com/artsy/force/blob/880f2132d53d993e5f1ac9afe5507c24f1226247/src/v2/Components/ArtworkFilter/ArtworkFilters/MediumFilter.tsx#L89). Definitely not something for right now, but maybe down the road.